File: pharmacy-system/src/main/java/com/example/pharmacysystem/service/AdminService.java
package com.example.pharmacysystem.service;

import com.example.pharmacysystem.model.UserInfo;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface AdminService {

    List<UserInfo> getUsers(int adminId);

    List<UserInfo> getAdmins(int adminId);

    boolean promoteUser(int adminId, int userId);

    UserInfo searchByUsername(int adminId, String username);
}

File: pharmacy-frontend/src/services/AdminService.js
import axios from 'axios';

const ADMIN_SERVICES_BASE_URL = `http://localhost:8088/admin`;

class AdminService {
    getUsers(adminId) {
        return axios.get(`${ADMIN_SERVICES_BASE_URL}/${adminId}/manageUsers`);
    }

    getAdmins(adminId) {
        return axios.get(`${ADMIN_SERVICES_BASE_URL}/${adminId}/getAdmins`);
    }

    searchByUsername(adminId, username) {
        return axios.get(`${ADMIN_SERVICES_BASE_URL}/searchUser/${adminId}/${username}`);
    }

    promoteUserToAdmin(adminId, userId) {
        return axios.put(`${ADMIN_SERVICES_BASE_URL}/promote-to-admin/${adminId}/${userId}`)
        .then(response => response.data)
        .catch(error => {
            throw error; // Handle the error appropriately
        });
    }
}

export default new AdminService();